lib/core/src/system/web/dom/html/html_renderer.rs

```diff
@@ -125,63 +125,62 @@ impl HTMLRendererData {
-        div   .set_property_or_panic("overflow"       , "hidden");
-        div   .set_property_or_panic("width"          , "100%");
-        div   .set_property_or_panic("height"         , "100%");
+        dom.set_property_or_panic("position", "absolute");
+        dom.set_property_or_panic("top"     , "0px");
+        dom.set_property_or_panic("overflow", "hidden");
+        dom.set_property_or_panic("overflow", "hidden");
+        dom.set_property_or_panic("width"   , "100%");
+        dom.set_property_or_panic("height"  , "100%");
         camera.set_property_or_panic("width"          , "100%");
         camera.set_property_or_panic("height"         , "100%");
         camera.set_property_or_panic("transform-style", "preserve-3d");
 
-        renderer.container.dom.append_or_panic(&div);
-        div                   .append_or_panic(&camera);
+        container.dom.append_or_panic(&dom);
+        dom.append_or_panic(&camera);
 
-        let data       = Rc::new(HTMLRendererData::new(div, camera));
-        let mut htmlrenderer = Self {renderer,data};
+        let data             = Rc::new(HTMLRendererData::new(dom,camera));
+        let mut htmlrenderer = Self {container,data};
 
         htmlrenderer.init_listeners();
         Ok(htmlrenderer)
     }
 
     fn init_listeners(&mut self) {
-        let dimensions = self.renderer.dimensions();
-        let data = self.data.clone();
-        self.renderer.add_resize_callback(move |dimensions:&Vector2<f32>| {
+        let dimensions = self.dimensions();
+        let data       = self.data.clone();
+        self.add_resize_callback(move |dimensions:&Vector2<f32>| {
             data.set_dimensions(*dimensions);
         });
         self.set_dimensions(dimensions);
     }
 
     /// Renders the `Scene` from `Camera`'s point of view.
-    pub fn render(&self, camera: &mut Camera, scene: &Scene<HTMLObject>) {
-        let trans_cam    = camera.transform().to_homogeneous().try_inverse();
-        let trans_cam    = trans_cam.expect("Camera's matrix is not invertible.");
-        let trans_cam    = trans_cam.map(eps);
-        let trans_cam    = invert_y(trans_cam);
-
-        let half_dim     = self.renderer.container.dimensions() / 2.0;
-        let y_scale      = camera.get_y_scale();
-        let y_scale      = y_scale * half_dim.y;
-
-        match camera.camera_type() {
-            CameraType::Perspective(_) => {
-                js::setup_perspective(&self.data.div, &y_scale.into());
+    pub fn render(&self, camera: &mut Camera2d, scene: &HTMLScene) {
+        camera.update();
+        let trans_cam  = camera.transform().matrix().try_inverse();
+        let trans_cam  = trans_cam.expect("Camera's matrix is not invertible.");
+        let trans_cam  = trans_cam.map(eps);
+        let trans_cam  = invert_y(trans_cam);
+        let half_dim   = self.container.dimensions() / 2.0;
+        let fov_slope  = camera.fovy_slope();
+        let y_scale    = fov_slope * half_dim.y;
```

Comment:
We need to describe it. It is named here `y_scale` it is passed to a function which argument is named `znear`. So `znear` is the same as `Y-axis` scale of something. I do not understand the logic. What is exactly this value responsible for? I'd be really thankful for an explanation.

Reply:
@wdanilo sorry to mix up names. `y_scale` is indeed meaningless. When we multiply `fov_slope` by half the screen height, we get the distance from which the camera can cover all the UI plane. The variable names were fixed.

Reply:
The trigonometric formula was fixed.